### PR TITLE
Fix various bugs with CI/CD workflows

### LINKED ISSUE #215: Fix create-automated-prs.yml, it does not trigger.
`create-automated-prs.yml` does not trigger on a `push` from another GitHub action. I can't remember where I saw it, but GH adds metadata to pushes from an action that causes other actions not to trigger from the action that pushed.

### LINKED ISSUE #217: ci.yml fails on docs branch push
When `ci.yml` executes, if there are previous, unmerged, changes to the `docs` branch and there are new changes to the `docs` on top of the previous unmerged changes there is a conflict and the `push` fails.

This likely has something to do with the `git checkout -b docs` command that creates a local docs branch without checking out the remote branch. One approach could be to check for the existence of the remote branch and if it exists just `git checkout docs` otherwise create the branch.

Another approach could be to test for the presence of the remote branch after creating the local branch and then if the remote branch exists do a `git pull origin docs`.

There are likely other ways to solve this problem.

### LINKED ISSUE #220: Fix cd-github-release.yml, it does not trigger
`cd-github-release.yml` is not triggered when `cd-create-version-tag.yml` creates a new tag. This is most likely because Github has built in functionality to avoid actions triggering other actions in order to avoid cyclic actions. How to work around this is to use a PAT token in the workflow that needs to trigger other workflows

- https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow

### PR DESCRIPTION
**Pull request by msha1026**
- To resolve #215:
  - Move automated PR creation step to `ci-module-versioning.yml` and `ci.yml`
  - Edit path trigger so that trigger works correctly
- To resolve #217:
  - Check if `version` or `docs` branch already exists in `ci-module-versioning.yml` and `ci.yml`, respectively. If they do, use existing branch
- To resolve #220:
  - Combine tagging main branch and creating a release into one workflow: `cd-version-tag-release.yml`
    - If we wanted to keep these as two separate workflows, we would need to create a PAT and store it as a secret in our project. Github doesn't allow workflows to "trigger" other workflows out of the box. In order to build something like that, we would need a PAT: https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow
  - Edit trigger -> PR is merged into main from the version branch

### CODE REVIEW

**Review by ozboms — APPROVED** [review]

File: .github/workflows/ci-module-versioning.yml [hunk 1]
```diff
@@ -31,17 +31,35 @@ jobs:
...
         run: |
-          git add ./Modules/BenchPress.Azure/BenchPress.Azure.psd1
-          git commit -m "version: updating version"
-          git push origin version
+          if [ "$(git diff --exit-code ./Modules/BenchPress.Azure/BenchPress.Azure.psd1 | wc -l)" -gt 0 ]
+          then
+            git add ./Modules/BenchPress.Azure/BenchPress.Azure.psd1
+            git commit -m "version: updating version"
+            git push origin version
+            if [ "$(gh pr list -H version -s open | wc -l)" -eq 0 ]
```
Comment (marclerwick) [inline .github/workflows/ci-module-versioning.yml:58]: I like this!

File: .github/workflows/cd-version-tag-release.yml [hunk 2]
```diff
@@ -24,3 +26,17 @@ jobs:
             | sed -n 's/\s*//p')
           git tag "$version"
           git push origin "$version"
+          echo "version=$version" >> "$GITHUB_OUTPUT"
+  create-release:
```
[the remark this answers is not in the document]
Comment (msha1026) [reply .github/workflows/cd-version-tag-release.yml:30]: So for the `Retrieve Version and Create Version Tag` step, we do not want to trigger a release after? Is that why the original implementation split out the github release into a separate workflow? We only want to create a Github release if a user creates a tag manually?

File: .github/workflows/cd-version-tag-release.yml [hunk 3]
```diff
@@ -2,18 +2,20 @@
...
-  push:
-    branches: [main]
   pull_request:
+    types: [closed]
     branches: [main]
 
 jobs:
   create-version-tag:
-    if: ${{ github.event_name == 'push' && github.event.pull_request.head.ref == 'refs/heads/version' }}
+    if: github.event.pull_request.merged == true && startsWith(github.head_ref, 'version')
```
[the remark this answers is not in the document]
Comment (msha1026) [reply .github/workflows/cd-version-tag-release.yml:11]: Technically either would work. I pulled it from this doc: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-based-on-the-head-or-base-branch-of-a-pull-request

But now I'm thinking it should just be a `==`. To make sure it only matches `version` branch and no other variations that could include the word `version`. Let me make that change.

File: .github/workflows/cd-version-tag-release.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (marclerwick) [reply .github/workflows/cd-version-tag-release.yml:30]: Ah, I see what you mean. Yeah, the "can't trigger from another action" would effectively disable the GH release workflow unless there was a manual version.

I think there was a conversation at one point about manually doing a GH release (which could be achieved with having a manual version tag only trigger), but I might be conflating that with the gated release to PS Gallery.

I can't think of a reason why the GH release shouldn't happen with each version.

[the remark this answers is not in the document]
Comment (msha1026) [reply .github/workflows/cd-version-tag-release.yml:30]: Ah sorry, I forgot to address the artifacts comment. The artifacts cannot be directories. When passing in directories, it errors out with the following:
```
Artifact is a directory:./BenchPress/Generators/. Directories can not be uploaded to a release.
```
It seems like it expects `.zip`. The action automatically zips the codebase and references it as an artifact for the release. Does this cover our needs, or do we want to explicitly zip the Module and Generator directories?

[the remark this answers is not in the document]
Comment (marclerwick) [reply .github/workflows/cd-version-tag-release.yml:30]: Interesting, individual files using wildcards work based on the example here: https://github.com/ncipollo/release-action

I wonder if the double star notation is unexpected.

File: .github/workflows/cd-version-tag-release.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (msha1026) [reply .github/workflows/cd-version-tag-release.yml]: Using `==` works as expected. Resolving.

File: .github/workflows/cd-version-tag-release.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (msha1026) [reply .github/workflows/cd-version-tag-release.yml:30]: Alright, I added a step to zip the Benchpress and Modules folders. Then use the zip as an artifact as a release.

**Review by marclerwick — APPROVED** [review]

**Pull request merged**